### PR TITLE
implementation

### PR DESCRIPTION
**Pull request by davidszotten**

### CODE REVIEW

File: test/conftest.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+# See http://eventlet.net/doc/patching.html#monkeypatching-the-standard-library
+import eventlet
+eventlet.monkey_patch()  # noqa (code before rest of imports)
+
+from nameko.containers import ServiceContainer
+import pytest
+
+
+@pytest.yield_fixture
+def container_factory():
```
Comment (mattbennett) [inline test/conftest.py:13]: Might be worth moving this function to nameko/testing/utils. Almost every application defines this fixture

File: test/test_nameko_socket_server.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    container = container_factory(Service, config)
+    container.start()
+
+    sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+    sock.connect(('', 6000))
+    sock.sendall('hello\n')
+    data = sock.recv(1024)
+    sock.close()
+    assert data == 'hello world'
+    container.stop()
```
Comment (mattbennett) [inline test/test_nameko_socket_server.py:31]: `container_factory` stops the container for you

File: test/conftest.py (hunk 1 above)
Comment (davidszotten) [inline test/conftest.py:13]: want to do that now?

File: test/test_nameko_socket_server.py (hunk 2 above)
Comment (davidszotten) [inline test/test_nameko_socket_server.py:31]: no, i changed it to `kill`. this exercises `stop`

Comment (mattbennett) [inline test/test_nameko_socket_server.py:31]: Ah ok.

**Comment by mattbennett** [comment]
:+1:

**Pull request merged**